packages/perseus/src/util/use-generate-url.ts

```diff
@@ -0,0 +1,42 @@
+import {createContext, useCallback, useContext} from "react";
+
+import {getDependencies} from "../dependencies";
+
+import {toAbsoluteUrl} from "./url-utils";
+
+import type {GenerateUrlArgs, GenerateUrlFn} from "../types";
+
+export const GenerateUrlContext = createContext<GenerateUrlFn | undefined>(
+    undefined,
+);
+
+export const useGenerateUrl = () => {
+    const staticUrl = getDependencies().staticUrl;
+
+    const generateUrl = useContext(GenerateUrlContext);
+
+    return useCallback(
+        (args: GenerateUrlArgs) => {
+            let updatedUrl = args.url;
+
+            switch (args.context) {
+                case "image_loader":
+                    updatedUrl = staticUrl(args.url);
+                    break;
+                case "python_program:program_url":
+                    updatedUrl = toAbsoluteUrl(args.url);
+                    break;
+            }
```

Comment:
@jeremywiebe I'm not sure if this is what you had in mind or not. We could isolate all calls to `staticUrl` and `toAbsoluteUrl` inside this function call, and deprecate those functions.